bigtable-dataflow-import/src/test/java/com/google/cloud/bigtable/dataflowimport/ImportIntegrationTest.java

```diff
@@ -0,0 +1,265 @@
+
+    assertCellSetsEqual(expected, tableAdmin.readAllCellsFromTable());
+    tableAdmin.dropTable();
+  }
+
+  private String getCloudTestDataFolder(boolean isHBase094Format) {
+    return cloudTestDataFolder + (isHBase094Format ? HBASE_094_SUBFOLDER : HBASE_CURRENT_SUBFOLDER);
+  }
+  /**
+   * Returns test data for import.
+   *
+   * <p>Currently for on-cloud tests, test sequence files are manually created and uploaded
+   * to Google Cloud Storage. If this method changes, the test files must be updated. The files
+   * currently in use are created as follows:
+   *
+   *<p>First, run the following commands in HBase shell:
+   * <pre>
+   *   create 'integration-test', { NAME => 'column_family', VERSIONS=>5}
+   *   put 'integration-test', 'row_key_1', 'column_family:col1', 'v1', 1
+   *   put 'integration-test', 'row_key_1', 'column_family:col1', 'v2', 2
+   *   put 'integration-test', 'row_key_2', 'column_family:col2', 'v3', 1
+   *   put 'integration-test', 'row_key_2', 'column_family:col2', 'v4', 3
+   * </pre>
+   *
+   * <p>Next, run the export command in the HBase home directory:
+   * <pre>
+   *   bin/hbase org.apache.hadoop.hbase.mapreduce.Export integration-test TARGET_DIR 5
+   * </pre>
+   */
+  private Set<? extends Cell> createTestData() throws Exception {
+    final byte[] rowKey1 = Bytes.toBytes("row_key_1");
+    final byte[] rowKey2 = Bytes.toBytes("row_key_2");
+    final byte[] col1 = Bytes.toBytes("col1");
+    final byte[] col2 = Bytes.toBytes("col2");
+    Set<? extends Cell> keyValues = Sets.newHashSet(
+        new KeyValue(rowKey1, CF, col1, 1L, Bytes.toBytes("v1")),
+        new KeyValue(rowKey1, CF, col1, 2L, Bytes.toBytes("v2")),
+        new KeyValue(rowKey2, CF, col2, 1L, Bytes.toBytes("v3")),
+        new KeyValue(rowKey2, CF, col2, 3L, Bytes.toBytes("v4")));
+    return keyValues;
+  }
+
+  private void doImport(HBaseImportOptions options) {
+    // Configure the TestPipeline for on- or off-cloud service.
+    if (options.getFilePattern().startsWith("gs://")) {
+      // Pipeline should be staged on cloud.
+      System.setProperty(RUN_INTEGRATION_TEST_ON_SERVICE, "true");
+      System.setProperty(TEST_PIPELINE_OPTIONS,
+          String.format("[ \"%s=%s\", \"%s=%s\", \"%s=%s\" ]",
+              "--runner", BlockingDataflowPipelineRunner.class.getSimpleName(),
+              "--project", options.getProject(),
+              "--stagingLocation", options.getStagingLocation()));
+    } else {
+      // Pipeline should run locally.
+      System.clearProperty(RUN_INTEGRATION_TEST_ON_SERVICE);
+    }
+
+    Pipeline p = CloudBigtableIO.initializeForWrite(TestPipeline.create());
+    p
+        .apply("ReadSequenceFile", Read.from(HBaseImportIO.createSource(options)))
+        .apply("ConvertResultToMutations", HBaseImportIO.transformToMutations())
+        .apply("WriteToTable", CloudBigtableIO.writeToTable(
+            CloudBigtableTableConfiguration.fromCBTOptions(options)));
+
+    p.run();
```

Comment:
What happen when a pipeline fails?
It is a good idea to catch some negative cases. e.g. a pipeline tries to read a file that is not there.